src/Potato/Data/Text/Zipper2.hs

```diff
@@ -140,13 +139,30 @@ rightWord = undefined
 rightLeftWord :: TextZipper -> TextZipper
 rightLeftWord = undefined
 
+-- | Clear the selection and move the cursor to the end of selection
+deselect :: TextZipper -> TextZipper
+deselect tz@(TextZipper lb b []           a la) = tz
+deselect    (TextZipper lb b [x]          a la) = TextZipper lb (b <> x) [] a la
+deselect    (TextZipper lb b (x:(xs:xss)) a la) = TextZipper ((reverse $ NE.init xs') <> [b <> x] <> lb) (NE.last xs') [] a la where
+                xs' = xs NE.:| xss
+
 -- | Move the cursor up one logical line (clearing the selection)
-up :: TextZipper -> TextZipper
-up z@(TextZipper lb b s a la) = undefined
+up ::  TextZipper -> TextZipper
+up    (TextZipper [] b [] a la) = TextZipper [] "" [] (b <> a) la
+up    (TextZipper [x] b [] a la) = TextZipper [] b' [] a' ([b <> a] <> la) where
+        (b', a') = T.splitAt (T.length b) x
+up    (TextZipper (x:xs) b [] a la) = TextZipper xs b' [] a' ([b <> a] <> la) where
+        (b', a') = T.splitAt (T.length b) x
+up tz@(TextZipper lb b s a la) = up $ deselect tz
 
 -- | Move the cursor down one logical line (clearing the selection)
 down :: TextZipper -> TextZipper
-down z@(TextZipper lb b s a la) = undefined
+down    (TextZipper lb b []  a []) = TextZipper lb (b <> a) [] "" []
+down    (TextZipper lb b [] a [x]) = TextZipper ([b <> a] <> lb) b' [] a' [] where
```

Comment:
this case is covered by `down    (TextZipper lb b [] a (x:xs))` case

Reply:
All review comments addressed